### ISSUE TITLE
Prune for GCXS

### ISSUE THREAD

**Issue by hameerabbasi** [pydata/sparse#408]
**Is your feature request related to a problem? Please describe.**
It would be nice to have a way to prune fill-values in `GCXS`.

**Describe the solution you'd like**
A `prune=` kwarg in the `GCXS` constructor.

**Additional context**
#407 is blocked on this.

**Comment by hameerabbasi** [comment]
Fixed by #409.
